source/dofs/dof_handler_policy.cc

```diff
@@ -990,17 +990,14 @@ namespace internal
          * Returns the final number of degrees of freedom, which is the number
          * of all valid DoF indices in @p new_dof_indices.
          */
-        template <int dim, int spacedim>
         static types::global_dof_index
         enumerate_dof_indices_for_renumbering(
           std::vector<types::global_dof_index> &new_dof_indices,
           const std::vector<
             std::map<types::global_dof_index, types::global_dof_index>>
-            &all_constrained_indices,
-          const DoFHandler<dim, spacedim> &)
+            &                           all_constrained_indices,
+          const types::global_dof_index start_dof_index)
         {
-          Assert(all_constrained_indices.size() == dim, ExcInternalError());
-
```

Comment:
why did this check go away?

Reply:
The only reason we had a `dim` template argument and pass in a `DoFHandler` was to ensure this size. Now I changed this to not pass in the unnecessary `DoFHandler` argument. It felt silly to pass in `dim` instead just to get the assertion working. As this is an internal interface I can't see that this assertion would help in detecting some bugs. If you want, I can add the check at the place we call this function.